### PR TITLE
nixos/minidlna: refactor

### PR DESCRIPTION
**Pull request by bb2020**
## Description of changes
#208242

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/networking/minidlna.nix [hunk 1]
```diff
@@ -1,41 +1,19 @@
...
+  options.services.minidlna.enable = lib.mkEnableOption "MiniDLNA, a simple DLNA server. Consider adding `openFirewall = true` into your config";
+  options.services.minidlna.openFirewall = lib.mkEnableOption "opening of both HTTP (TCP) and SSDP (UDP) ports in the firewall";
 
   options.services.minidlna.settings = lib.mkOption {
     default = {};
-    description = ''
-      The contents of MiniDLNA's configuration file.
-      When the service is activated, a basic template is generated from the current options opened here.
-    '';
+    description = "";
```
Comment (aanderse) [inline nixos/modules/services/networking/minidlna.nix:14]: why blank?

File: nixos/modules/services/networking/minidlna.nix [hunk 2]
```diff
@@ -106,29 +88,17 @@ in
     };
   };
 
-  imports = [
```
Comment (aanderse) [inline nixos/modules/services/networking/minidlna.nix:110]: no reason to remove these yet, is there? we have instances much older than this

File: nixos/modules/services/networking/minidlna.nix [hunk 3]
```diff
@@ -88,6 +65,11 @@ in
         example = "general,artwork,database,inotify,scanner,metadata,http,ssdp,tivo=warn";
         description = "Defines the type of messages that should be logged and down to which level of importance.";
       };
+      options.enable_subtitles = lib.mkOption {
```
Comment (aanderse) [inline nixos/modules/services/networking/minidlna.nix:68]: this should be in a separate commit because it isn't a refactor

File: nixos/modules/services/networking/minidlna.nix [hunk 4]
```diff
@@ -1,41 +1,19 @@
...
-  options.services.minidlna.openFirewall = lib.mkOption {
-    type = lib.types.bool;
-    default = false;
-    description = ''
-      Whether to open both HTTP (TCP) and SSDP (UDP) ports in the firewall.
-    '';
-  };
+in {
+  options.services.minidlna.enable = lib.mkEnableOption "MiniDLNA, a simple DLNA server. Consider adding `openFirewall = true` into your config";
+  options.services.minidlna.openFirewall = lib.mkEnableOption "opening of both HTTP (TCP) and SSDP (UDP) ports in the firewall";
```
Comment (aanderse) [inline nixos/modules/services/networking/minidlna.nix:10]: i don't think the grammar checks out like you think it does here... 🤔

File: nixos/modules/services/networking/minidlna.nix (hunk 2 above)
Comment (bb2020) [reply nixos/modules/services/networking/minidlna.nix:110]: I presume this module is not widely used anyway. Let's clean it up and hopefully not touch it again.

**Review by fpletz — APPROVED** [review]
